pkg/discovery/worker/dispatcher_test.go

```diff
@@ -55,3 +69,75 @@ func TestDispatcher_DispatchMimic(t *testing.T) {
 		t.Errorf("%d Vs. %d", receiveNum, nodeNum)
 	}
 }
+
+func getDispatcherAndCollector(workerCount, taskRunTimeSec int) (*Dispatcher, *ResultCollector) {
+	clusterScraper := &cluster.ClusterScraper{}
+	probeConfig := &configs.ProbeConfig{
+		MonitoringConfigs: []monitoring.MonitorWorkerConfig{&monitoring.DummyMonitorConfig{
+			TaskRunTime: taskRunTimeSec,
+		}},
+	}
+	dispatcherConfig := NewDispatcherConfig(clusterScraper, probeConfig, workerCount, 10)
+	dispatcher := NewDispatcher(dispatcherConfig)
+	resultCollector := NewResultCollector(workerCount * 2)
+	dispatcher.Init(resultCollector)
+	return dispatcher, resultCollector
+}
+
+// Test dispatching 1000 tasks with 100 workers, each worker run for 1 second, expect the test to finish in 10 seconds
```

Comment:
This is a safe assumption here because the workers are doing nothing but sleep. However having 100 workers would be meaningless if there is say only 1 cpu available.